### ISSUE TITLE
Exiting Docker Desktop UI also closes any open terminal windows

### ISSUE THREAD

**Issue by random-pixels** [docker/desktop-linux#109]
Issue is that when exiting the Docker Desktop UI by right-clicking on the icon in the system notification area, terminal windows are also closed (even if you are actively typing in them or they are producing output at the time).

I can't find any option within Docker Desktop to disable this behaviour, so I assume it's an unintended (and frustrating) bug.

This happens on a fresh install of Docker Desktop with no images, volumes, containers, etc as well as an updated Docker Desktop as well as Docker Desktop containing 2-3 images, containers and one volume.

  - [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
- Docker Desktop UI exits. Terminal windows stay where they are.
### Actual behavior
- Docker Desktop UI exits. At the same time, terminal windows close suddenly and unexpectedly.
### Information

  - Linux distro: Linux Mint (with Cinnamon desktop environment)
  - Distro version: 21.0 Vanessa
  - Docker Desktop Version: 4.15.0 (but also affecting 4.14.x and 4.11.x)
  - Note: This is a native Linux system, not a VM running in Windows or anything, in case that matters

### Steps to reproduce the behavior

I don't think you need any Docker images, any special settings, modifications, extensions as this issue happens straight out of the box.
Docker is not logged in with an account or connected to anything remote.
Also, it does not matter if the terminal window(s) were opened before or after Docker Desktop.

  1. Open a terminal command window(s)
  2. Open Docker Desktop
  3. Close Docker Desktop (which minimizes it to the system notification area)
  4. Exit Docker Desktop by right-clicking on its icon in the notification area
  5. Observe that the terminal command window closes at the same instant as Docker Desktop exits

**Comment by tsoud** [comment]
Exact same issue here, with same steps. Linux Mint (21.1 - Vera).

**Comment by rmwiseman** [comment]
> Exact same issue here, with same steps. Linux Mint (21.1 - Vera). Docker Desktop v4.16.2

Exact same issue, steps, Mint version and Docker Desktop version here too.

**Comment by jacksontong** [comment]
> Exact same issue here, with same steps. Linux Mint (21.1 - Vera). Docker Desktop v4.16.2

Same issue.

**Comment by blueschu** [comment]
Same issue here. Right-clicking on the Docker Desktop icon in the system tray and clicking "Quit Docker Desktop" results in all open gnome-terminal windows being closed at the same time as the Docker Desktop UI.

It seems that Docker Desktop is killing the `gnome-terminal-server` process on exit. 

Reproduced in 
* Linux Mint 21.0-Vanessa and 21.1-Vera (both Cinnamon DE)
* fresh installs of Docker Desktop v4.19.0
* gnome-terminal v3.44.0.

As a mitigation, I discovered that Docker Desktop only kills the default `gnome-terminal-server` process that is spawned in the background by `gnome-terminal`. If you start `gnome-terminal-server` manually, it survives Docker Desktop shutdown.

### Mitigation Steps

1. Kill any existing instance of `gnome-terminal-server`. This can be done by closing Docker Desktop per this issue, or by killing the process manually, e.g. by finding its PID with `ps aux | grep gnome-terminal-server` and then  `kill <pid>` . **This will kill any open gnome terminals windows**.
2. Start `gnome-terminal-server` ***without opening a terminal.*** In Cinnamon, this can be done by typing `ALT+F2` and entering `/usr/libexec/gnome-terminal-server` (the full path to the gnome-terminal-server executable).
    1. Note: if you just start `gnome-terminal` from the ALT+F2 menu, the background gnome-terminal-server process that it creates will still not survive Docker Desktop shutdown.
3. Open terminals like normal. All new `gnome-terminal` windows should now connect to the `gnome-terminal-server` process you started, and should now survive Docker Desktop shutdown.

**Comment by rmunn** [comment]
Still occurring with Docker Desktop version 4.21.1. All reports of this so far (including me) have been using Linux Mint 21; perhaps this behavior is specific to Mint and/or the Cinnamon desktop environment. The workaround from https://github.com/docker/desktop-linux/issues/109#issuecomment-1544438060 has been working for me, but it shouldn't really be necessary: Docker Desktop shouldn't be killing processes it didn't start.

**Comment by fmichel** [comment]
Still occurring with 4.24.2, on Archlinux with lxde. So it is not specific to Mint and pretty annoying...

**Comment by pimisen** [comment]
Still an issue in 4.27!
Also on Debian 12 with cinnamon. VERY anoying!

**Comment by rmunn** [comment]
Still occurring with version 4.30.0-149282. The workaround from https://github.com/docker/desktop-linux/issues/109#issuecomment-1544438060 (start `gnome-terminal-server` manually first, and only **then** start `docker-desktop`) continues to work.

**Comment by vitali-yanushchyk-valor** [comment]
Still occurring. v.4.34.2 (167172)

**Comment by rmunn** [comment]
Still occurring with version 4.35.1-173168 of the docker-desktop-amd64.deb package.

As an aside, this is why I prefer to use open-source tools. It's been nearly two years since this issue was reported, and the Docker Desktop devs (who are quite busy with other tasks) haven't gotten around to looking into this issue yet, or even saying anything on the issue thread. With an open-source tool, I could dive into the code, figure out where it's killing gnome-terminal-server, and figure out how to make Docker Desktop check first whether that process is the one that it started itself, rather than blindly (and rudely) killing a process that it didn't start.

But since this tool isn't open source, all I can do is keep commenting when new versions come out without this bug being addressed. It's a little frustrating that no communication at all has happened. Docker Desktop dev team, I know quite well how busy you can be, but this bug is both high on the "user annoyance" factor and high on the "easy to fix" factor, which makes it low-hanging fruit. Please at least say something in the comments so we know you're aware of the problem and looking into it.

And yes, I'm fully aware that it might turn out to actually not be easy to fix. But since I can't know for certain, as this tool is not open-source, all I can do is guess. And my guess, based on previous experience with other bugs in other tools, is that this would be fairly straightforward to fix for anyone with intermediate-to-expert skills with Linux. Surely you must have some of those people on your team. We'd love to hear from them.

**Comment by Danita** [comment]
@akerouanton Yes please, a word from the developers would be greatly appreciated.

**Comment by gdbonino** [comment]
This issue is almost two years old. It is really annoying that every time I close Docker Desktop all terminals close.

**Comment by LukeWale** [comment]
Same issue still

Linux Mint 22.1 Cinnamon 6.4.8
Standard terminal
Docker desktop 4.34.3 (170107)

**Comment by Danita** [comment]
Any word about this? It's really annoying!

**Comment by GKPeiris** [comment]
Facing same issue here
Linux Mint 21.3 Virginia
    base: Ubuntu 22.04 jammy
Docker Desktop 4.43.2

**Comment by gdbonino** [comment]
It doesn't seem to be something that anyone is interested in fixing :(

**Comment by rmunn** [comment]
One possible workaround, since this seems to be a low priority for the Docker team to fix, would be to switch to a different terminal emulator: this only seems to affect `gnome-terminal` because that's what Docker Desktop is using to provide the external terminal (when you click on "Open in external terminal" in a container's Exec tab). I switched to Ghostty (https://ghostty.org/) and have not experienced this bug any more. In fact, gnome-terminal is not installed on my new laptop, so clicking "Open in external terminal" produces the error "gnone-terminal not found" and the feature doesn't work.

(Which also explains why Docker Desktop closes gnome-terminal windows on exit, because it thinks that it opened them and is responsible for cleaning them up — but the "should I close this terminal window?" logic is incorrect and over-broad).

**Comment by noah-foltz** [comment]
Confirming this is still bugged

**Comment by gdbonino** [comment]
Surely another technology will come out to replace Docker before they fix this bug :(

**Comment by Sadere** [comment]
Not fixed yet

**Comment by yihune21** [comment]
Not fixed yet 
Docker version 29.5.2
Linux mint VERSION="22.3 (Zena)"

**Comment by tntscreed** [comment]
Still an issue to this day. Docker Desktop 4.84.0, Docker 29.7.1
